antd/src/main/scala/demo/App.scala

```diff
@@ -428,24 +439,46 @@ object CSS extends js.Any
     )
 
     val renderRangePicker = section(
-      h2("Range Picker"), {
-        val (currentValue, setValue) =
-          useState[RangeValue[Moment]] {
-            val endMoment = moment()
-            val startMoment = endMoment.subtract(DurationConstructor.hours, 2)
-            // need type descriptions to help Scala infer that there can be `| Null` at two levels here
-            js.Tuple2(startMoment: EventValue[Moment], endMoment: EventValue[Moment])
-          }
+      h2("Range Picker"),
+      DatePicker.PickerBaseProps.RangePicker
+        .RangePickerDateProps()
+        .showTime(RangeShowTimeObject[Moment].setFormat("HH:mm"))
+        .format("YYYY/MM/DD HH:mm")
+        .value(rangePickerValues)
+        .onChange { (values: RangeValue[Moment], formatString: js.Tuple2[String, String]) =>
+          console.log(formatString)
+          updateRangePickerValues(values)
+        }
+    )
 
-        DatePicker.PickerBaseProps.RangePicker
-          .RangePickerDateProps()
-          .showTime(RangeShowTimeObject[Moment].setFormat("HH:mm"))
-          .format("YYYY/MM/DD HH:mm")
-          .value(currentValue)
-          .onChange { (values: RangeValue[Moment], formatString: js.Tuple2[String, String]) =>
-            console.log(formatString)
-            setValue(values)
-          }
+    val renderTreeSelect = section(
```

Comment:
I refactored it slightly, how about this?

```scala
    val renderTreeSelect = section(
      h2("Multiple and checkable Tree Select"), {
        def node(title: String, value: String) =
          DataNode().setTitle(title).setValue(value).setKey(value)

        val data: js.Array[DataNode] = js.Array(
          node("Node1", "0-0").setChildrenVarargs(
            node("Child Node1", "0-0-0")
          ),
          node("Node2", "0-1").setChildrenVarargs(
            node("Child Node3", "0-1-0"),
            node("Child Node4", "0-1-1"),
            node("Child Node5", "0-1-2")
          )
        )

        TreeSelect[js.Array[String]]
          .value(selectTreeValues)
          .onChange((values, _, _) => updateSelectTreeValues(values))
          .treeData(data)
          .placeholder("Please select")
          .treeCheckable(true)
          .showCheckedStrategy(strategyUtilMod.SHOW_PARENT)
          .style(CSSProperties().setWidth("100%"))
      }
    )
```

Reply:
Much cleaner, thank you :+1: